bot/exts/fun/anagram.py

```diff
@@ -0,0 +1,97 @@
+import asyncio
+import json
+import logging
+import random
+from pathlib import Path
+
+import discord
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours, NEGATIVE_REPLIES
+
+log = logging.getLogger(__name__)
+
+TIME_LIMIT = 60
+
+# Issue with the json file , gotta try it on local environment first
+ANAGRAMS_ALL = json.loads(Path("bot/resources/fun/anagram.json").read_text("utf8"))
+
+
+class Anagram(commands.Cog):
+    """Cog for the Anagram game command."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.winners = set()
+        self.correct = ""
+        self.current_channel = None
+
+    @commands.command(name="anagram", aliases=("anag", "gram", "ag"))
+    async def anagram_command(self, ctx: commands.Context) -> None:
+        """
+        Given shuffled letters which can be rearranged to form Anagrams.
+
+        If this works correctly will update this more in detail
+        """
+        if self.current_channel:
+            await ctx.send(f"An anagram is already being solved in {self.current_channel.mention}!")
+            return
+
+        # Don't let users start in a DM
+        if not ctx.guild:
+            await ctx.send(
+                embed=discord.Embed(
+                    title=random.choice(NEGATIVE_REPLIES),
+                    description="You can't start riddles in DMs",
+                    colour=discord.Colour.red()
+                )
+            )
+            return
+
+        self.current_channel = ctx.channel
+
+        anagram_letters, anagram_answers = random.choice(list(ANAGRAMS_ALL.items()))
+        scrambled_letters = anagram_letters
+        self.correct = anagram_answers
+
+        embed_title = f"Find anagrams from these letters '{scrambled_letters.upper()}'"
+        embed_description = f"You have {TIME_LIMIT} seconds to find all words."
```

Comment:
This claims that the objective is to find all words, but the code says otherwise. Because you win if you guess one right word, not if you as a team find all of them.

Reply:
Have updated the text so "all" is not mentioned, looking at the json file it seemed very hard for people to get all the anagrams correctly, so to reduce complexity and make it easier for people, even if someone mentions a correct anagram I have added them to the winners list